lib/MusicBrainz/Server/Data/Search.pm

```diff
@@ -319,11 +319,17 @@ sub search
             SQL
 
         $use_hard_search_limit = 0;
-    } elsif ($type eq 'tag') {
+    }
+
+    elsif ($type eq 'tag') {
         $query = <<~'SQL';
-            SELECT tag.id, tag.name, genre.id AS genre_id,
+            SELECT tag.id,
+                   tag.name,
+                   coalesce(genre.id, genre_alias.genre) AS genre_id,
                    ts_rank_cd(mb_simple_tsvector(tag.name), query, 2) AS rank
-            FROM tag LEFT JOIN genre USING (name), plainto_tsquery('mb_simple', mb_lower(?)) AS query
+            FROM tag
+            LEFT JOIN genre USING (name)
+            LEFT JOIN genre_alias USING (name), plainto_tsquery('mb_simple', mb_lower(?)) AS query
             WHERE mb_simple_tsvector(tag.name) @@ query
```

Comment:
Does this need a GROUP BY now, since there might be different genre aliases sharing the same name?

Reply:
We should never reuse aliases since that would break the system anyway if we want to use them to count the tags for votes - although I guess this is not enforced right now other than by me not adding dupes